pkg/api/dirs.go

```diff
@@ -0,0 +1,149 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package api
+
+import (
+	"archive/tar"
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"mime"
+	"net/http"
+	"path/filepath"
+	"strings"
+
+	"github.com/ethersphere/bee/pkg/jsonhttp"
+	"github.com/ethersphere/bee/pkg/logging"
+	"github.com/ethersphere/bee/pkg/manifest/jsonmanifest"
+	"github.com/ethersphere/bee/pkg/storage"
+	"github.com/ethersphere/bee/pkg/swarm"
+)
+
+// dirUploadInfo contains the data for a directory to be uploaded
+type dirUploadInfo struct {
+	toEncrypt bool
+	reader    io.ReadCloser
+}
+
+// dirUploadHandler uploads a directory supplied as a tar in an HTTP request
+func (s *server) dirUploadHandler(w http.ResponseWriter, r *http.Request) {
+	dirInfo, err := getDirHTTPInfo(r)
```

Comment:
what about checking if the `Content-Type` header indicates that the request is of type tar? [link](https://stackoverflow.com/questions/39058006/proper-mime-media-type-for-tar-files)

Reply:
done.

also added a test case for this.